lib/rucio/daemons/tracer/kronos.py

```diff
@@ -403,17 +403,13 @@ def kronos_dataset(once: bool = False, dataset_queue: Queue = None, sleep_time:
         )
     )
 
-    # once again for the backlog
+    # once again for the any backlog
     __update_datasets(dataset_queue)
 
 
 def run_once_kronos_dataset(heartbeat_handler: HeartbeatHandler, dataset_queue: Queue, **kwargs):
     _, _, logger = heartbeat_handler.live()
-    dataset_wait = config_get_int('tracer-kronos', 'dataset_wait')
-    start = datetime.now()
-    if (datetime.now() - start).seconds > dataset_wait:
-        __update_datasets(dataset_queue=dataset_queue, logger=logger)
-        start = datetime.now()
+    __update_datasets(dataset_queue=dataset_queue, logger=logger)
```

Comment:
is there any point to have this two-line function ? merge __update_Datasets into this one?

Reply:
I could return the `heartbeat_handler` from the merged method and re-use when it is called again manually. Might look hacky though

Reply:
I have a better way to do this. Sending the updated PR ;)

Reply:
I see now. But how does this protect us from any race conditions ? nothing is stopping the AMQ consumer from continuing to consume the queue while we call the second _update_datasets. So, should we call it a 3rd time, a 4th, ... ?

Reply:
I agreel. it does not, so I won't defend this. ;)

Reply:
I looked though the code and it was probably useful, because that second call is done just after the "sleep". So it reduces the number of unhandled messages. Sorry for the confusion. Probably a good idea to keep the second call for now, but we must find a better solution for all these AMQ consumer daemons .

Reply:
Yeah sure, but doing it a 3rd, or 4rth time would reduce the number of unhandled messages even further :P. I will add it back to the code